src/TimerDeviceCountup.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System;
+using Crestron.SimplSharp;
+using Crestron.SimplSharpPro.DeviceSupport;
+using PepperDash.Core;
+using PepperDash.Essentials.Core;
+using PepperDash.Essentials.Core.Bridges;
+
+namespace TimerDevice
+{
+    public class CountupTimer : EssentialsBridgeableDevice
+    {
+        private CTimer _countupCTimer;
+        private TimeSpan _countupTimerTimeSpan;
+        private bool _autoStopOnStartReleaseBool;
+        public BoolFeedback CountupTimerRunningFb { get; private set; }
+        public StringFeedback CountupTimerValueFb { get; private set; }
+        public event EventHandler<CountupTimerEventArgs> CountupTimerChanged;
+        public bool IsRunning { get; private set; }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="key"></param>
+        public CountupTimer(string key, string name, CountupTimerPropertiesConfig propertiesConfig)
+            : base(key, name)
+        {
+            if (propertiesConfig == null) return;
+
+            _autoStopOnStartReleaseBool = propertiesConfig.autoStopOnStartRelease;
+            CountupTimerRunningFb = new BoolFeedback(() => this.IsRunning);
+            CountupTimerValueFb = new StringFeedback(() => _countupTimerTimeSpan.ToString());
+
+            CrestronEnvironment.ProgramStatusEventHandler += eventType =>
+            {
+                if (eventType != eProgramStatusEventType.Stopping)
+                {
+                    this.Stop();
+                    _countupCTimer = null;
+                }
+            };
+        }
+
+        /// <summary>
+        /// Update the method signature to use the custom EventArgs class:
+        /// </summary>
+        /// <param name="e"></param>
+        protected virtual void OnCountupTimerChange(CountupTimerEventArgs e)
+        {
+            CountupTimerChanged += (sender, args) => { };
+        }
+
+        /// <summary>
+        /// Method to trigger custom event
+        /// </summary>
+        /// <param name="message"></param>
+        public void TriggerCountupTimerChange(string message)
+        {
+            OnCountupTimerChange(new CountupTimerEventArgs(message));
+        }
+        
+        /// <summary>
+        /// Start countupTimer
+        /// </summary>
+        public void Start()
+        {
+            if (IsRunning) return;
+            IsRunning = true;
+            CountupTimerRunningFb.FireUpdate();
+
+            Debug.Console(1, this, "CountupTimer.Start() requested...");
+
+            if(_countupTimerTimeSpan == null)
```

Comment:
TimeSpans are structs, so this will never be null (structs are never null)

Reply:
Commit [42518c4 ](https://github.com/PepperDash/epi-utilities-timer/commit/42518c4edff1f9d8ca654d93eeff439de855df30) removes the null check, adds Unit-Test logic for future reference, and updates the bridge IP ID to match Unit-Test.